### DISCUSSION TITLE
Package updates publishing error, package searching error for some keywords

### DISCUSSION THREAD

**Discussion by staniska** [pulsar-edit/.github#119] [Community Packages]
I've created and published package for Atom named "sinumerik-highlight". I tried to publish some updates to it shortly before Atom was archived, but was unable to do so due to getting an error "Creating new version failed: GitHub API returned error: TooManyRequests". Later I tried to remove my package from the Atom registry and publish it again but couldn't do that either. 

Now searching on https://web.pulsar-edit.dev/packages/search for the keyword "sinumerik" fails with the error "There was an error on this page! Sorry!", as well as a second page with search results for keyword "highlight"

Trying to update package "sinumerik-highlight" with command "pulsar -p publish patch" publishes a new tag to my repository (https://github.com/staniska/sinumerik-highlight), then fails:
"Creating new version failed: The package.json at owner/repo isn't valid."

Can anyone help me to remove this package from the Pulsar so I can re-publish it?

P.S. Sorry for my English

**Comment by confused-Techie** [comment]
@staniska This seems related to a bug which we thought was resolved on the backend previously, but obviously still has a means of being seen.

Essentially there was a short period of time were a bug would affect packages having a new version published that would omit some needed fields for a package to be published. And when this happened essentially the package wouldn't work very nicely with the frontend site causing the error you are seeing when your package appears in search results.

What this means is that while I would be more than happy to work with you to help remove the package from the registry and allow you to republish you don't have to. As this is an error on our end I'd be more than happy to manually make the fixes myself, or at least get it to a state where you are able to publish that same version over what currently exists on the Registry.

What may be the smoothest choice as of now, is we are in the works for some pretty major updates to the backend that will likely resolve this bug (As well as others) and once those are rolled out to production then we would be in a fantastic point, confidence wise, to get your package totally updated.

But otherwise if you would still like to have your package totally removed feel free to let me know and we can work that out. But I should warn you that once removed the name you previously published under is still taken, so you would need to rename it if you planned to publish the same package again.

Either way happy to find what works best for you, and sorry for the confusion around it.

P.S. No worries, totally fantastic English!

**Comment by staniska** [comment]
@confused-Techie Thank you for fast response.
I don't need to publish updates as soon as possible, so I'll wait for backend updates and then try again.

**Reply by confused-Techie** [reply]
Fantastic! I'll try to ping you back, but otherwise keep a lookout on our Discord announcements, or otherwise here are some PRs to keep your eyes on (And we will probably tag a new release for the backend after these updates)

* The one that should stop this issue from being reintroduced: [PR#50](https://github.com/pulsar-edit/package-backend/pull/50)
* [PR65](https://github.com/pulsar-edit/package-backend/pull/65)
* [PR66](https://github.com/pulsar-edit/package-backend/pull/66)

**Reply by staniska** [reply]
Hello.
I've read the sources of Pulsar & Pulsar-backend I think "The package.json at owner/repo isn't valid" error reason is wrong data in backend database. Can you send me result object from the function db.getPackageByName("sinumerik-highlight", true)  (src/handlers/post_package_handler.js   str.292) to a3boot.babaev@gmail.com so I can see more details of the error?

**Reply by confused-Techie** [reply]
@staniska You are 100% correct. And I really do apologize about that. The metadata for your package is incorrect. During the very early days of the Pulsar backend, there for a short time existed a bug that could corrupt the data present in this location. There were attempts to manually and automatically resolve it, but it seems your package never was identified as having an error.

I'll go ahead and try to resolve this tomorrow to get the data to be proper. I really do apologize about that, and appreciate your patience and willingness to work with us on this

**Reply by staniska** [reply]
@confused-Techie Thanks, I really appreciate your help.
